src/app/donation-start/donation-start-form/donation-tipping-slider/donation-tipping-slider.component.ts

```diff
@@ -0,0 +1,214 @@
+import { AfterContentInit, Component, ElementRef, Input, OnChanges, OnDestroy, OnInit, Renderer2, ViewChild } from '@angular/core';
+
+@Component({
+  selector: 'app-donation-tipping-slider',
+  templateUrl: './donation-tipping-slider.component.html',
+  styleUrls: ['./donation-tipping-slider.component.scss']
+})
+export class DonationTippingSliderComponent implements OnInit, AfterContentInit, OnChanges, OnDestroy {
+
+
+  @Input() spaceBelow: 0 | 1 | 2 | 3 | 4 | 5 | 6 = 0;
+
+  @Input() colourScheme:
+  | 'primary' // blue
+  | 'secondary' // green
+  | 'tertiary' // coral
+  | 'brand-1' // cc-red
+  | 'brand-2' // wgmf-purple
+  | 'brand-3' // gmf-green
+  | 'brand-4' // emf-yellow
+  | 'brand-5' // c4c-orange
+  | 'brand-6' // mhf-turquoise
+  | 'white'
+  | 'black'
+  | 'grey-extra-light'
+  | 'grey-light'
+  | 'grey-medium'
+  | 'grey-dark' = 'primary';
+
+  @Input() percentageCurrent: number;
+  @Input() percentageStart: number;
+  @Input() percentageEnd: number;
+  @Input() donationAmount: number;
+  //ISO-4217 currency code (e.g. GBP, USD)
+  @Input() donationCurrency!: 'GBP' | 'USD';
+  @Input() onHandleMoved: (tipPercentage: number, tipAmount: number) => void;
+
+  @ViewChild('handle', {static: true}) handle: ElementRef;
+  @ViewChild('bar', {static: true}) bar: ElementRef;
+  @ViewChild('percentageValue', {static: true}) percentageWrap: ElementRef;
+  @ViewChild('donationValue', {static: true}) donationWrap: ElementRef;
+
+
+  containerClass: string;
+  derivedPercentage: number;
+  tipAmount: number;
+  currencyFormatted: string;
+
+  documentMouseupUnlistener: () => void;
+  documentMousemoveUnlistener: () => void;
+  documentTouchmoveUnlistener: () => void;
+
+  isMoving = false;
+  max: number;
+  pageX: any;
+  mousePos: number;
+  position: number;
+  disableDefaults: boolean = false;
+
+  constructor(
+    public renderer: Renderer2
+  ) {
+    this.documentMouseupUnlistener = renderer.listen('document', 'mouseup', () => {
+      this.isMoving = false;
+    });
+
+    this.documentMousemoveUnlistener = renderer.listen('document', 'mousemove', (event: MouseEvent | TouchEvent) => {
+      this.move(event);
+    });
+
+    this.documentTouchmoveUnlistener = renderer.listen('document', 'touchmove', (event: MouseEvent | TouchEvent) => {
+      this.move(event);
+    });
+  };
+
+
+  ngOnInit() {
+    this.containerClass = 'container space-below-' + this.spaceBelow;
+    this.calcAndSetPercentage();
+    this.adjustDonationPercentageAndValue();
+    this.updateHandlePosition(undefined)
+  }
+
```

Comment:
What's the reasoning for doing some stuff OnInit and waiting for AfterContentInit for other stuff? I guess the nativeElements aren't ready to have event listeners OnInit?